### DISCUSSION TITLE
@remix-pwa/push - Cannot find package 'web-push'

### DISCUSSION THREAD

**Discussion by kunuva** [ShafSpecs/remix-pwa#148] [General]
I receive this error when trying to use usePush() - "Internal server error: Cannot find package 'web-push' imported from node_modules\@remix-pwa\push\dist\server\notifications.js"

**Comment by ShafSpecs** [comment]
> I receive this error when trying to use usePush() - "Internal server error: Cannot find package 'web-push' imported from node_modules\@remix-pwa\push\dist\server\notifications.js"

Can you try installing web push itself as a dependency? And are you importing use push from /client?

**Reply by Tiny-House-Tech** [reply]
I did get it working by installing web-push as a dependency, I had to update @remix-pwa/push/dist/server/notifications.js - line 1 = "import * as webpush from 'web-push'" to "import webpush from 'web-push'" to get it work. I thought I'd let you know it doesn't work out of the box and see if there was a fix because it won't work from a clean npm install.

**Reply by ShafSpecs** [reply]
Thanks. Would look at that this weekend

**Reply by kunuva** [reply]
Sorry, I logged into my other PC and accidentally replied on a different account. Thanks for all the work you've put into remix-pwa, it's a great package.

**Reply by ShafSpecs** [reply]
> I did get it working by installing web-push as a dependency, I had to update @remix-pwa/push/dist/server/notifications.js - line 1 = "import * as webpush from 'web-push'" to "import webpush from 'web-push'" to get it work. I thought I'd let you know it doesn't work out of the box and see if there was a fix because it won't work from a clean npm install.

I didn't install web-push within the push package so the user needed to install it themselves. I have fixed that as well as the broken web-push import. Publishing a patch in a few minutes, lemme know if any other thing comes up 😁
